src/App.tsx

```diff
@@ -14,11 +14,89 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {
+  selectedGood: string[];
+};
+
+class App extends React.Component<{}, Props> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  addGoods = (goods: string) => {
+    if (!this.state.selectedGood.includes(goods)) {
+      this.setState((state => ({
+        selectedGood: [...state.selectedGood, goods],
+      })));
+    }
+  };
+
+  deleteGoods = () => {
+    this.setState({ selectedGood: [] });
+  };
+
+  aboutProducts = () => {
+    const { selectedGood } = this.state;
+
+    if (selectedGood.length === 1) {
+      return `${selectedGood} is selected`;
+    }
+
+    if (selectedGood.length > 1) {
+      return `${selectedGood.join(', ')} are selected`;
+    }
+
+    return 'No goods selected';
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="list__selected">
+          Selected good:
+          {' '}
```

Comment:
Тут тебе попросять переробити на Template string.  або зроби так  що  ти повертав тільки {this.aboutProducts()}

а aboutProducts  хай повертає той напис що тобі потрібно

```suggestion
          {this.aboutProducts()}
```

Reply:
виклик функції aboutProducts потрібно перемістити між render i return а тут використати шиблонні літерали
```
{`Selected good: ${your_variable}`}
```

Reply:
а хіба ти на сьогоднішній лекції не казав, що не варто вставляти функції між render i return?